dts/arm/nxp/nxp_rt5xx_common.dtsi

```diff
@@ -0,0 +1,487 @@
+/*
+ * Copyright (c) 2022, NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <mem.h>
+#include <arm/armv8-m.dtsi>
+#include <dt-bindings/clock/mcux_lpc_syscon_clock.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/i2c/i2c.h>
+
+/ {
+	chosen {
+		zephyr,entropy = &trng;
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			compatible = "arm,cortex-m33f";
+			reg = <0>;
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			mpu: mpu@e000ed90 {
+				compatible = "arm,armv8m-mpu";
+				reg = <0xe000ed90 0x40>;
+				arm,num-mpu-regions = <8>;
+			};
+		};
+	};
+};
+
+&sram {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	sram0: memory@20180000 {
+		compatible = "mmio-sram";
+		reg = <0x20180000 DT_SIZE_K(3072)>;
+	};
+
+	sram1: memory@40140000 {
+		compatible = "mmio-sram";
+		reg = <0x40140000 DT_SIZE_K(16)>;
+	};
+
+};
+
+&systick {
+	/*
+	 * RT500 relies by default on the OS Timer for system clock
+	 * implementation, so the SysTick node is not to be enabled.
+	 */
+	status = "disabled";
+};
+
+&peripheral {
+	#address-cells = <1>;
+	#size-cells = <1>;
+	/*
+	 * Note that the offsets here are relative to the base address
+	 * defined in either nxp_rt5xx_ns.dtsi or nxp_rt5xx.dtsi. The base
+	 * addresses differ between non-secure (0x40000000) and secure
+	 * modes (0x50000000).
+	 */
+
+	clkctl0: clkctl@1000 {
+		compatible = "nxp,lpc-syscon";
+		reg = <0x1000 0x1000>;
+		label = "CLKCTL_0";
+		#clock-cells = <1>;
+	};
+
+	clkctl1: clkctl@21000 {
+		compatible = "nxp,lpc-syscon";
+		reg = <0x21000 0x1000>;
+		label = "CLKCTL_1";
+		#clock-cells = <1>;
+	};
+
+	uuid: uuid@2f50 {
+		compatible = "nxp,lpc-uid";
+		reg = <0x2f50 0x10>;
+	};
+
+	gpio0: gpio@0 {
+		compatible = "nxp,lpc-gpio";
+		reg = <0x100000 0x1000>;
+		interrupts = <4 2>,<5 2>,<6 2>,<7 2>;
+		label = "GPIO_0";
+		gpio-controller;
+		#gpio-cells = <2>;
+		port = <0>;
+	};
+
+	gpio1: gpio@1 {
+		compatible = "nxp,lpc-gpio";
+		reg = <0x100000 0x1000>;
+		interrupts = <35 2>,<36 2>,<37 2>,<38 2>;
+		label = "GPIO_1";
+		gpio-controller;
+		#gpio-cells = <2>;
+		port = <1>;
+	};
+
+	gpio2: gpio@2 {
+		compatible = "nxp,lpc-gpio";
+		reg = <0x100000 0x1000>;
+		label = "GPIO_2";
+		gpio-controller;
+		#gpio-cells = <2>;
+		port = <2>;
+	};
+
+	gpio3: gpio@3 {
+		compatible = "nxp,lpc-gpio";
+		reg = <0x100000 0x1000>;
+		label = "GPIO_3";
+		gpio-controller;
+		#gpio-cells = <2>;
+		port = <3>;
+	};
+
+	gpio4: gpio@4 {
+		compatible = "nxp,lpc-gpio";
+		reg = <0x100000 0x1000>;
+		label = "GPIO_4";
+		gpio-controller;
+		#gpio-cells = <2>;
+		port = <4>;
+	};
+
+	gpio5: gpio@5 {
+		compatible = "nxp,lpc-gpio";
+		reg = <0x100000 0x1000>;
+		label = "GPIO_5";
+		gpio-controller;
+		#gpio-cells = <2>;
+		port = <5>;
+	};
+
+	gpio6: gpio@6 {
+		compatible = "nxp,lpc-gpio";
+		reg = <0x100000 0x1000>;
+		label = "GPIO_6";
+		gpio-controller;
+		#gpio-cells = <2>;
+		port = <6>;
+	};
```

Comment:
question: why do all GPIOs share the same reg? what is the point of `gpio@N` then?

Reply:
The GPIO on this part is located at 0x<n>010_0000 where <n> is the offset for secure/ns access. The access to the various GPIO instances is via a port offset written into the register where we have values 0 through 6 for this part. The same thing is present in the rt6xx and other LPC based IP blocks.